pkg/management/component.go

```diff
@@ -789,6 +793,76 @@ func (c *component) GetRolesOfUser(ctx context.Context, realmName, userID string
 	return rolesRep, nil
 }
 
+func (c *component) rolesToMap(roles []kc.RoleRepresentation) map[string]kc.RoleRepresentation {
+	var res = make(map[string]kc.RoleRepresentation)
+	for _, role := range roles {
+		res[*role.ID] = role
+	}
+	return res
+}
+
+func (c *component) getRolesAsMap(ctx context.Context, realmName string) (map[string]kc.RoleRepresentation, error) {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+	var roles, err = c.keycloakClient.GetRoles(accessToken, realmName)
+	if err != nil {
+		c.logger.Info(ctx, "msg", "Failed to get realm role mappings", "realm", realmName, "user")
+		return nil, err
+	}
+	return c.rolesToMap(roles), nil
+}
+
+func (c *component) getUserRolesAsMap(ctx context.Context, realmName, userID string) (map[string]kc.RoleRepresentation, error) {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+	var roles, err = c.keycloakClient.GetRealmLevelRoleMappings(accessToken, realmName, userID)
+	if err != nil {
+		c.logger.Info(ctx, "msg", "Failed to get user roles", "realm", realmName, "user")
+		return nil, err
+	}
+	return c.rolesToMap(roles), nil
+}
+
+func (c *component) AddRoleToUser(ctx context.Context, realmName, userID, roleID string) error {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+	var rolesMap, err = c.getRolesAsMap(ctx, realmName)
+	if err != nil {
+		return err
+	}
+	var roles []kc.RoleRepresentation
+	if role, ok := rolesMap[roleID]; ok {
+		roles = append(roles, role)
+	}
+	if len(roles) == 0 {
+		c.logger.Info(ctx, "msg", "Unknown role", "realm", realmName, "user", userID, "role", roleID)
+		return errorhandler.CreateBadRequestError("role")
+	}
+	err = c.keycloakClient.AddRealmLevelRoleMappings(accessToken, realmName, userID, roles)
+	if err != nil {
+		c.logger.Info(ctx, "msg", "Failed to add user roles", "user", userID, "role", roleID)
+	}
+	return err
+}
+
+func (c *component) DeleteRoleForUser(ctx context.Context, realmName, userID, roleID string) error {
```

Comment:
I think the 2 previous comments also applies here.

Reply:
I think my 2 previous answers also applies here. ;-)